Samples/PowerShell/ADLSUser/Set-AdlsAccess.ps1

```diff
@@ -100,29 +128,52 @@ function copyacls
         $pathToSet = Join-Path -Path $Path -ChildPath $item.PathSuffix;
         $pathToSet = $pathToSet.Replace("\", "/");
         
-        if ($Permissions -ieq "All")
+        if ($item.Type -ieq "FILE")
         {
-            $perms = "rwx";
+            # set the ACL without default
+            giveaccess -Account $Account -Path $pathToSet -Id $Id -EntityType $EntityType -Permissions $Permissions | Out-Null
         }
-        elseif ($Permissions -ieq "ReadExecute")
+        elseif ($item.Type -ieq "DIRECTORY")
         {
-            $perms = "r-x";
+            # set permission and recurse on the directory
+            giveaccess -Account $Account -Path $pathToSet -Id $Id -EntityType $EntityType -Permissions $Permissions -IsDefault | Out-Null
+            setacerec -Account $Account -Path $pathToSet -Permissions $Permissions -Id $Id -EntityType $EntityType | Out-Null
         }
         else
```

Comment:
minor: you could get rid of this else since there are only two valid item types in the system (file and directory). This is just super extra safe :)